diff -ur cfitsio-4.7.0.orig/drvrsmem.h cfitsio-4.7.0/drvrsmem.h
--- cfitsio-4.7.0.orig/drvrsmem.h	2026-08-14 18:53:45.601516300 -0400
+++ cfitsio-4.7.0/drvrsmem.h	2026-08-14 18:54:38.895719690 -0400
@@ -8,9 +8,11 @@
 */
 
 
+#ifndef _WIN32
 #include <sys/ipc.h>		/* this is necessary for Solaris/Linux */
 #include <sys/shm.h>
 #include <sys/sem.h>
+#endif
 
 #ifdef _AIX
 #include <fcntl.h>
diff -ur cfitsio-4.7.0.orig/Makefile.am cfitsio-4.7.0/Makefile.am
--- cfitsio-4.7.0.orig/Makefile.am	2026-08-14 18:53:45.599728368 -0400
+++ cfitsio-4.7.0/Makefile.am	2026-08-14 18:56:03.554980855 -0400
@@ -134,7 +134,6 @@
 	fpack \
 	funpack \
 	imcopy \
-	smem \
 	speed
 
 cookbook_SOURCES = utilities/cookbook.c
@@ -166,7 +165,7 @@
 
 if !NOFORTRAN
 
-noinst_PROGRAMS+= testf77
+#noinst_PROGRAMS+= testf77
 testf77_SOURCES = utilities/testf77.f
 
 endif
diff -ur cfitsio-4.7.0.orig/Makefile.in cfitsio-4.7.0/Makefile.in
--- cfitsio-4.7.0.orig/Makefile.in	2026-08-14 18:53:45.613026477 -0400
+++ cfitsio-4.7.0/Makefile.in	2026-08-14 18:57:08.227708197 -0400
@@ -97,10 +97,10 @@
 @NOFORTRAN_FALSE@am__append_1 = $(F77_WRAPPERS)
 @NOGSIFTP_FALSE@am__append_2 = ${GSIFTP_SRC}
 bin_PROGRAMS = cookbook$(EXEEXT) fitscopy$(EXEEXT) fitsverify$(EXEEXT) \
-	fpack$(EXEEXT) funpack$(EXEEXT) imcopy$(EXEEXT) smem$(EXEEXT) \
+	fpack$(EXEEXT) funpack$(EXEEXT) imcopy$(EXEEXT) \
 	speed$(EXEEXT)
 noinst_PROGRAMS = testprog$(EXEEXT) $(am__EXEEXT_2) $(am__EXEEXT_3)
-@NOFORTRAN_FALSE@am__append_3 = testf77
+#@NOFORTRAN_FALSE@am__append_3 = testf77
 @BUILD_ITER_TRUE@am__append_4 = iter_image iter_var
 TESTS = run-testprog $(am__EXEEXT_1) tests/coverage-test
 check_PROGRAMS = $(am__EXEEXT_1)
@@ -149,7 +149,7 @@
 	tests/test_region$(EXEEXT) tests/test_ricecomp$(EXEEXT) \
 	tests/test_scalnull$(EXEEXT) tests/test_simplerng$(EXEEXT) \
 	tests/test_wcssub$(EXEEXT) tests/test_wcsutil$(EXEEXT)
-@NOFORTRAN_FALSE@am__EXEEXT_2 = testf77$(EXEEXT)
+#@NOFORTRAN_FALSE@am__EXEEXT_2 = testf77$(EXEEXT)
 @BUILD_ITER_TRUE@am__EXEEXT_3 = iter_image$(EXEEXT) iter_var$(EXEEXT)
 PROGRAMS = $(bin_PROGRAMS) $(noinst_PROGRAMS)
 am__vpath_adj_setup = srcdirstrip=`echo "$(srcdir)" | sed 's|.|.|g'`;
