diff -ur cfitsio-4.6.4.orig/drvrsmem.h cfitsio-4.6.4/drvrsmem.h
--- cfitsio-4.6.4.orig/drvrsmem.h	2026-05-09 16:42:20.138903870 -0400
+++ cfitsio-4.6.4/drvrsmem.h	2026-05-09 16:43:08.129095252 -0400
@@ -8,9 +8,11 @@
 */
 
 
+#ifndef _WIN32
 #include <sys/ipc.h>		/* this is necessary for Solaris/Linux */
 #include <sys/shm.h>
 #include <sys/sem.h>
+#endif
 
 #ifdef _AIX
 #include <fcntl.h>
diff -ur cfitsio-4.6.4.orig/Makefile.am cfitsio-4.6.4/Makefile.am
--- cfitsio-4.6.4.orig/Makefile.am	2026-05-09 16:42:20.091902703 -0400
+++ cfitsio-4.6.4/Makefile.am	2026-05-09 16:46:06.870532607 -0400
@@ -134,7 +134,6 @@
 	fpack \
 	funpack \
 	imcopy \
-	smem \
 	speed
 
 cookbook_SOURCES = utilities/cookbook.c
@@ -166,7 +165,7 @@
 
 if !NOFORTRAN
 
-noinst_PROGRAMS+= testf77
+#noinst_PROGRAMS+= testf77
 testf77_SOURCES = utilities/testf77.f
 
 endif
diff -ur cfitsio-4.6.4.orig/Makefile.in cfitsio-4.6.4/Makefile.in
--- cfitsio-4.6.4.orig/Makefile.in	2026-05-09 16:42:20.092902727 -0400
+++ cfitsio-4.6.4/Makefile.in	2026-05-09 16:46:57.162781138 -0400
@@ -97,10 +97,10 @@
 @NOFORTRAN_FALSE@am__append_1 = $(F77_WRAPPERS)
 @NOGSIFTP_FALSE@am__append_2 = ${GSIFTP_SRC}
 bin_PROGRAMS = cookbook$(EXEEXT) fitscopy$(EXEEXT) fitsverify$(EXEEXT) \
-	fpack$(EXEEXT) funpack$(EXEEXT) imcopy$(EXEEXT) smem$(EXEEXT) \
+	fpack$(EXEEXT) funpack$(EXEEXT) imcopy$(EXEEXT) \
 	speed$(EXEEXT)
 noinst_PROGRAMS = testprog$(EXEEXT) $(am__EXEEXT_1) $(am__EXEEXT_2)
-@NOFORTRAN_FALSE@am__append_3 = testf77
+#@NOFORTRAN_FALSE@am__append_3 = testf77
 @BUILD_ITER_TRUE@am__append_4 = iter_image iter_var
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -120,7 +120,7 @@
 CONFIG_CLEAN_VPATH_FILES =
 am__installdirs = "$(DESTDIR)$(bindir)" "$(DESTDIR)$(libdir)" \
 	"$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(includedir)"
-@NOFORTRAN_FALSE@am__EXEEXT_1 = testf77$(EXEEXT)
+#@NOFORTRAN_FALSE@am__EXEEXT_1 = testf77$(EXEEXT)
 @BUILD_ITER_TRUE@am__EXEEXT_2 = iter_image$(EXEEXT) iter_var$(EXEEXT)
 PROGRAMS = $(bin_PROGRAMS) $(noinst_PROGRAMS)
 am__vpath_adj_setup = srcdirstrip=`echo "$(srcdir)" | sed 's|.|.|g'`;
