Include correct header to check for Win32 threads.

diff -urN fftw-3.3.10/configure.ac.orig fftw-3.3.10/configure.ac
--- fftw-3.3.10/configure.ac.orig	2021-09-14 16:39:39.000000000 +0200
+++ fftw-3.3.10/configure.ac	2026-04-06 13:07:48.948924055 +0200
@@ -699,7 +699,7 @@
         # Win32 threads are the default on Windows:
     if test -z "$THREADLIBS"; then
         AC_MSG_CHECKING([for Win32 threads])
-        AC_TRY_LINK([#include <windows.h>],
+        AC_TRY_LINK([#include <process.h>],
             [_beginthreadex(0,0,0,0,0,0);],
             [THREADLIBS=" "; AC_MSG_RESULT(yes)],
             [AC_MSG_RESULT(no)])
