diff -ur flac-1.4.1.orig/configure.ac flac-1.4.1/configure.ac
--- flac-1.4.1.orig/configure.ac	2022-09-24 11:02:18.098776155 -0400
+++ flac-1.4.1/configure.ac	2022-09-24 11:06:34.016159903 -0400
@@ -376,8 +376,12 @@
 AM_CONDITIONAL(FLaC__HAS_DOXYGEN, test -n "$DOXYGEN")
 
 if test  "x$DOXYGEN" = "x"; then
-	AC_CHECK_FILE($srcdir/doc/FLAC.tag,[HAVE_PREBUILT_FLAC_TAG=yes])
-	AC_CHECK_FILE($srcdir/doc/api/modules.html,[HAVE_PREBUILT_API_DIR=yes])
+	if test -f "$srcdir/doc/FLAC.tag"; then
+          HAVE_PREBUILT_FLAC_TAG=yes
+	fi
+	if test -f "$srcdir/doc/api/modules.html"; then
+          HAVE_PREBUILT_API_DIR=yes
+	fi
 fi
 AM_CONDITIONAL(FLaC__HAS_PREBUILT_DOXYGEN, test "x$HAVE_PREBUILD_FLAG_TAG$HAVEPREBUILD_API_DIR" = "xyesyes")
 
@@ -483,8 +487,12 @@
 AC_DEFINE(FLAC__HAS_PANDOC)
 AH_TEMPLATE(FLAC__HAS_PANDOC, [define if you have pandoc])
 else
-AC_CHECK_FILE($srcdir/man/flac.1,[HAVE_MAN_FLAC=yes])
-AC_CHECK_FILE($srcdir/man/metaflac.1,[HAVE_MAN_METAFLAC=yes])
+  if test -f "$srcdir/man/flac.1"; then
+     HAVE_MAN_FLAC=yes
+  fi
+  if test -f "$srcdir/man/metaflac.1"; then
+     HAVE_MAN_METAFLAC=yes
+  fi
 fi
 AM_CONDITIONAL(FLaC__HAS_PREBUILT_MANPAGES, test "x$HAVE_MAN_FLAC$HAVE_MAN_METAFLAC" = "xyesyes")
 
