diff -ur jasper-version-4.1.1.orig/CMakeLists.txt jasper-version-4.1.1/CMakeLists.txt
--- jasper-version-4.1.1.orig/CMakeLists.txt	2023-12-18 08:52:34.109693740 -0500
+++ jasper-version-4.1.1/CMakeLists.txt	2023-12-18 08:53:34.680118322 -0500
@@ -789,10 +789,12 @@
 # Check for the Math library.
 ################################################################################
 
+if(NOT WIN32)
 find_library(MATH_LIBRARY m)
 if(NOT MATH_LIBRARY)
 	set(MATH_LIBRARY "")
 endif()
+endif()
 
 ################################################################################
 # Perform shared library setup.
diff -ur jasper-version-4.1.1.orig/src/libjasper/CMakeLists.txt jasper-version-4.1.1/src/libjasper/CMakeLists.txt
--- jasper-version-4.1.1.orig/src/libjasper/CMakeLists.txt	2023-12-18 08:52:34.121694418 -0500
+++ jasper-version-4.1.1/src/libjasper/CMakeLists.txt	2023-12-18 08:54:36.677623584 -0500
@@ -188,7 +188,7 @@
 	target_compile_definitions(libjasper PRIVATE _CRT_NONSTDC_NO_DEPRECATE)
 endif()
 
-if(UNIX)
+if(UNIX OR MINGW)
 	set_target_properties(libjasper PROPERTIES
 	  VERSION ${JAS_SO_NAME} SOVERSION ${JAS_SO_VERSION})
 endif()
diff -ur jasper-version-4.1.1.orig/src/libjasper/include/jasper/jas_stream.h jasper-version-4.1.1/src/libjasper/include/jasper/jas_stream.h
--- jasper-version-4.1.1.orig/src/libjasper/include/jasper/jas_stream.h	2023-12-18 08:52:34.123694531 -0500
+++ jasper-version-4.1.1/src/libjasper/include/jasper/jas_stream.h	2023-12-18 08:58:41.750479720 -0500
@@ -258,6 +258,10 @@
 typedef struct {
 	int fd;
 	int flags;
+#ifndef _UCRT
+#undef L_tmpnam
+#define L_tmpnam FILENAME_MAX
+#endif
 #if defined(JAS_WASI_LIBC)
 #define L_tmpnam 4096
 #endif
diff -ur jasper-version-4.1.1.orig/src/libjasper/jp2/jp2_cod.h jasper-version-4.1.1/src/libjasper/jp2/jp2_cod.h
--- jasper-version-4.1.1.orig/src/libjasper/jp2/jp2_cod.h	2023-12-18 08:52:34.123694531 -0500
+++ jasper-version-4.1.1/src/libjasper/jp2/jp2_cod.h	2023-12-18 08:56:24.907742782 -0500
@@ -288,10 +288,10 @@
 * Box class.
 \******************************************************************************/
 
-jp2_box_t *jp2_box_create(int type);
-void jp2_box_destroy(jp2_box_t *box);
-jp2_box_t *jp2_box_get(jas_stream_t *in);
-int jp2_box_put(jp2_box_t *box, jas_stream_t *out);
+JAS_DLLEXPORT jp2_box_t *jp2_box_create(int type);
+JAS_DLLEXPORT void jp2_box_destroy(jp2_box_t *box);
+JAS_DLLEXPORT jp2_box_t *jp2_box_get(jas_stream_t *in);
+JAS_DLLEXPORT int jp2_box_put(jp2_box_t *box, jas_stream_t *out);
 
 JAS_ATTRIBUTE_CONST
 static inline uint_least8_t JP2_DTYPETOBPC(uint_least8_t dtype)
@@ -309,6 +309,6 @@
 #define ICC_CS_YCBCR	0x59436272
 #define ICC_CS_GRAY	0x47524159
 
-const jp2_cdefchan_t *jp2_cdef_lookup(jp2_cdef_t *cdef, int channo);
+JAS_DLLEXPORT const jp2_cdefchan_t *jp2_cdef_lookup(jp2_cdef_t *cdef, int channo);
 
 #endif
