diff -ur libxml2-v2.9.13.orig/libxml-2.0.pc.in libxml2-v2.9.13/libxml-2.0.pc.in
--- libxml2-v2.9.13.orig/libxml-2.0.pc.in	2022-02-23 13:29:48.119724801 -0500
+++ libxml2-v2.9.13/libxml-2.0.pc.in	2022-02-23 13:42:19.580206424 -0500
@@ -8,6 +8,7 @@
 Version: @VERSION@
 Description: libXML library version2.
 Requires:
+Requires.private: liblzma
 Libs: -L${libdir} -lxml2
 Libs.private: @ICU_LIBS@ @THREAD_LIBS@ @Z_LIBS@ @LZMA_LIBS@ @ICONV_LIBS@ @M_LIBS@ @WIN32_EXTRA_LIBADD@ @LIBS@
 Cflags: @XML_INCLUDEDIR@ @XML_CFLAGS@
