diff -ur mesa-21.0.3.orig/src/gallium/auxiliary/gallivm/lp_bld_tgsi_soa.c mesa-21.0.3/src/gallium/auxiliary/gallivm/lp_bld_tgsi_soa.c
--- mesa-21.0.3.orig/src/gallium/auxiliary/gallivm/lp_bld_tgsi_soa.c	2021-05-20 18:45:28.645574035 -0400
+++ mesa-21.0.3/src/gallium/auxiliary/gallivm/lp_bld_tgsi_soa.c	2021-05-20 18:50:13.771317137 -0400
@@ -305,7 +305,7 @@
    LLVMBuilderRef builder = mask->bld->gallivm->builder;
    struct function_ctx *ctx = func_ctx(mask);
 
-   int default_exec_pc;
+   int default_exec_pc = 0;
    boolean default_is_last;
 
    if (ctx->switch_stack_size > LP_MAX_TGSI_NESTING) {
diff -ur mesa-21.0.3.orig/src/util/u_string.h mesa-21.0.3/src/util/u_string.h
--- mesa-21.0.3.orig/src/util/u_string.h	2021-05-20 18:45:28.898588502 -0400
+++ mesa-21.0.3/src/util/u_string.h	2021-05-25 08:35:19.002123122 -0400
@@ -42,6 +42,8 @@
 #include <stddef.h>
 #include <stdarg.h>
 #include <string.h>
+#include <limits.h>
+
 
 #include "util/macros.h" // PRINTFLIKE
 
@@ -72,7 +74,7 @@
 {
    va_list ap;
    va_start(ap, format);
-   vsnprintf(str, (size_t)-1, format, ap);
+   vsnprintf(str, INT_MAX, format, ap);
    va_end(ap);
 }
 
