From a3b60fc23f2968e2d22c8d46baf29d4cd90b0d5c Mon Sep 17 00:00:00 2001
From: Jeremy Drake <github@jdrake.com>
Date: Sat, 16 Oct 2021 16:50:46 -0700
Subject: [PATCH] mesa: fixes for UCRT.

--- mesa-21.2.4/src/compiler/nir/nir.h.orig	2021-10-14 12:59:05.367845000 -0700
+++ mesa-21.2.4/src/compiler/nir/nir.h	2021-10-16 15:55:09.881268400 -0700
@@ -54,7 +54,7 @@
 
 #include "nir_opcodes.h"
 
-#if defined(_WIN32) && !defined(snprintf)
+#if defined(_WIN32) && !defined(snprintf) && !defined(_UCRT)
 #define snprintf _snprintf
 #endif
 
