diff -ur cfitsio-4.1.0.orig/configure.in cfitsio-4.1.0/configure.in
--- cfitsio-4.1.0.orig/configure.in	2022-05-26 08:43:32.205001801 -0400
+++ cfitsio-4.1.0/configure.in	2022-05-26 08:44:53.020552448 -0400
@@ -489,7 +489,7 @@
 if test "x$ADD_CURL" = xyes; then
   # Use curl-config to get compiler & linker flags, if available.
   AC_CHECK_PROG([CURLCONFIG], [curl-config], [curl-config], [], [], [])
-  if test "x$CURLCONFIG" != x; then
+  if test "x$CURLCONFIG" != x && test "x$CURLCONFIG" != "xnone"; then
     CURL_LIB=`$CURLCONFIG --libs`
     CURL_INC=`$CURLCONFIG --cflags`
     if test "x$CURL_LIB" != x; then
