diff -ru units-2.25.orig/units.c units-2.25/units.c
--- units-2.25.orig/units.c	2026-02-16 13:38:51.036107168 -0500
+++ units-2.25/units.c	2026-02-16 13:39:16.751402880 -0500
@@ -723,7 +723,7 @@
     free(widestr);
     return -1; /* invalid multibyte sequence */
   }
-#  if defined _WIN32 && defined _MSC_VER
+#  if defined _WIN32
   len=wcslen(widestr);
 #  else
   len=wcswidth(widestr, len);
@@ -4380,7 +4380,7 @@
 {
    int lines = 20;      /* arbitrary but probably safe value */
 
-#if defined (_WIN32) && defined (_MSC_VER)
+#if defined (_WIN32)
    CONSOLE_SCREEN_BUFFER_INFO csbi;
 
    if (GetConsoleScreenBufferInfo(GetStdHandle(STD_OUTPUT_HANDLE), &csbi))
@@ -4802,7 +4802,7 @@
   char *progdir = NULL;
   char *p;
 
-#if defined (_WIN32) && defined (_MSC_VER)
+#if defined (_WIN32)
   char buf[FILENAME_MAX + 1];
 
   /* get the full pathname of the current executable and be done with it */
@@ -7090,7 +7090,7 @@
 checklocale()
 {
   char *temp;
-#if defined _WIN32 && defined _MSC_VER
+#if defined _WIN32
   /* only works in Windows Terminal */
   if (getenv(WINDOWS_TERMINAL)) {
       utf8mode = (GetConsoleCP() == CP_UTF8)
