diff -ur nsis-code-7405-1-NSIS-trunk.orig/SCons/Config/gnu nsis-code-7405-1-NSIS-trunk/SCons/Config/gnu
--- nsis-code-7405-1-NSIS-trunk.orig/SCons/Config/gnu	2024-02-15 10:57:22.343383597 -0500
+++ nsis-code-7405-1-NSIS-trunk/SCons/Config/gnu	2024-02-15 11:00:48.067246323 -0500
@@ -54,7 +54,7 @@
 # Don't allow mingw to link with LIBGCC*.DLL and LIBSTDC++-*.DLL
 def configure_static_libs(env):
 	env.Append(LINKFLAGS = ['-static-libgcc'])
-	env.Append(LINKFLAGS = ['-static-libstdc++']) # MinGW GCC 4.5.2 warns about unrecognized option but it also actually needs it!
+	#env.Append(LINKFLAGS = ['-static-libstdc++']) # MinGW GCC 4.5.2 warns about unrecognized option but it also actually needs it!
 
 if defenv['PLATFORM'] == 'win32':
 	configure_static_libs(defenv)
