diff -ur interval-3.2.2.orig/src/Makefile interval-3.2.2/src/Makefile
--- interval-3.2.2.orig/src/Makefile	2026-02-18 08:11:16.692120074 -0500
+++ interval-3.2.2/src/Makefile	2026-02-18 08:12:30.774523975 -0500
@@ -33,6 +33,8 @@
 CONF_FLAG_SSE2 ?=--enable-sse2
 endif
 
+CRLIBM_CONFIG_FLAGS ?=
+
 .PHONY: all
 all: $(OBJ)
 
@@ -55,7 +57,7 @@
 $(BUNDLED_CRLIBM_DIR)/crlibm_private.o: $(BUNDLED_CRLIBM_DIR)/crlibm_config.h
 	$(MAKE) -C $(BUNDLED_CRLIBM_DIR) CFLAGS+="$(shell $(MKOCTFILE) -p CPICFLAG) -Wno-div-by-zero -Wno-unused-variable -Wno-unused-but-set-variable"
 $(BUNDLED_CRLIBM_DIR)/crlibm_config.h:
-	(cd $(BUNDLED_CRLIBM_DIR) && ./configure --disable-dependency-tracking $(CONF_FLAG_SSE2))
+	(cd $(BUNDLED_CRLIBM_DIR) && ./configure --disable-dependency-tracking $(CONF_FLAG_SSE2) $(CRLIBM_CONFIG_FLAGS))
 
 ## GNU MPFR api oct-files
 mpfr_matrix_mul_d.oct mpfr_matrix_sqr_d.oct : mpfr_%.oct: mpfr_%.cc mpfr_commons.h compatibility/octave.h compatibility/mpfr.h
