diff -ur pstoedit-3.77.orig/src/callgs.cpp pstoedit-3.77/src/callgs.cpp
--- pstoedit-3.77.orig/src/callgs.cpp	2021-09-06 16:18:29.786833687 -0400
+++ pstoedit-3.77/src/callgs.cpp	2021-09-06 16:23:10.927193819 -0400
@@ -368,7 +368,14 @@
 		} else {
 		    if (verbose) errstream<< "nothing found in gsview32.ini file - using find_gs to lookup latest version of Ghostscript in registry " << endl;
 			static char buf[1000];
-			if (find_gs(buf, sizeof(buf), 550 /* min ver*/ , getPstoeditsetDLLUsage() , gsregbase, verbose)) { 
+
+	                gstocall = getenv("GS");
+
+                        if(gstocall) {
+		                if (verbose)
+			                errstream << "GS is set to:" << gstocall << endl;
+
+                        } else if (find_gs(buf, sizeof(buf), 550 /* min ver*/ , getPstoeditsetDLLUsage() , gsregbase, verbose)) { 
 				if (verbose) {
 					(void)dumpgsvers(gsregbase, verbose);
 					if (getPstoeditsetDLLUsage()) errstream << "Latest GS DLL is " << buf << endl;
