diff -ur pstoedit-3.77.orig/othersrc/gsdllinc/wgsver.c pstoedit-3.77/othersrc/gsdllinc/wgsver.c
--- pstoedit-3.77.orig/othersrc/gsdllinc/wgsver.c	2021-09-08 15:38:09.812856463 -0400
+++ pstoedit-3.77/othersrc/gsdllinc/wgsver.c	2021-09-08 15:39:08.695322622 -0400
@@ -40,6 +40,9 @@
 #include "wgsver.h"
 #include "cppcomp.h"
 
+#ifndef KEY_WOW64_64KEY
+#define KEY_WOW64_64KEY 0x0100
+#endif
 /* Ghostscript may be known in the Windows Registry by
  * the following names.
  */
diff -ur pstoedit-3.77.orig/src/cbstream.cpp pstoedit-3.77/src/cbstream.cpp
--- pstoedit-3.77.orig/src/cbstream.cpp	2021-09-08 15:38:09.813856522 -0400
+++ pstoedit-3.77/src/cbstream.cpp	2021-09-08 15:39:08.696322681 -0400
@@ -33,6 +33,8 @@
 // this code is only needed under WIN32
 // ...and under OS/2
 
+#include <cstdio>
+
 #include "cppcomp.h"
 
 #include "cbstream.h"
diff -ur pstoedit-3.77.orig/src/dynload.cpp pstoedit-3.77/src/dynload.cpp
--- pstoedit-3.77.orig/src/dynload.cpp	2021-09-08 15:38:09.817856757 -0400
+++ pstoedit-3.77/src/dynload.cpp	2021-09-08 15:42:20.225597212 -0400
@@ -241,7 +241,7 @@
 #ifdef _WIN64
 	DynLoader::fptr rfptr = /* ptr_to_fptr */(GetProcAddress((HINSTANCE) handle, name));	//lint !e611 //: Suspicious cast
 #else
-	DynLoader::fptr rfptr = ptr_to_fptr(GetProcAddress((HINSTANCE)handle, name));
+	DynLoader::fptr rfptr = ptr_to_fptr((void*)GetProcAddress((HINSTANCE)handle, name));
 #endif
 
 #else
diff -ur pstoedit-3.77.orig/src/Makefile.am pstoedit-3.77/src/Makefile.am
--- pstoedit-3.77.orig/src/Makefile.am	2021-09-08 15:38:09.817856757 -0400
+++ pstoedit-3.77/src/Makefile.am	2021-09-08 15:39:08.696322681 -0400
@@ -144,7 +144,8 @@
 	psfront.h				\
 	genericints.h
 
-libpstoedit_la_LIBADD = ${CXX_STD_LIB} ${CXX_RUNTIME_LIB} -ldl -lm
+libpstoedit_la_LIBADD = ${CXX_STD_LIB} ${CXX_RUNTIME_LIB} -lm
+libpstoedit_la_CPPFLAGS = -I$(top_srcdir)/othersrc/gsdllinc
 libpstoedit_la_LDFLAGS = -no-undefined
 
 pstoedit_SOURCES = cmdmain.cpp
@@ -171,7 +172,7 @@
 pstoedit.cpp : pstoedit.ph
 
 pstoedit.ph : pstoedit.pro
-	awk -f ps_pro_to_ph.awk < pstoedit.pro > pstoedit.ph
+	awk -f $(srcdir)/ps_pro_to_ph.awk < $(srcdir)/pstoedit.pro > pstoedit.ph
 
 cppcheck:
 	cppcheck --inconclusive --enable=all `sh ./get_D_and_I_options.sh ${AM_CPPFLAGS}` ${SOURCES}
