diff -ur QScintilla_src-2.13.0.orig/src/qscintilla.pro QScintilla_src-2.13.0/src/qscintilla.pro
--- QScintilla_src-2.13.0.orig/src/qscintilla.pro	2021-10-16 14:29:45.588604124 -0400
+++ QScintilla_src-2.13.0/src/qscintilla.pro	2021-10-16 14:30:15.765967678 -0400
@@ -21,7 +21,7 @@
 !win32:VERSION = 15.1.0
 
 TEMPLATE = lib
-CONFIG += qt warn_off thread exceptions hide_symbols
+CONFIG += qt warn_off release thread exceptions
 
 CONFIG(debug, debug|release) {
     mac: {
