Don't check for libclang when building without it.

diff -urN qttools-everywhere-src-6.10.1/configure.cmake.orig qttools-everywhere-src-6.10.1/configure.cmake
--- qttools-everywhere-src-6.10.1/configure.cmake.orig	2025-11-13 21:37:00.000000000 +0100
+++ qttools-everywhere-src-6.10.1/configure.cmake	2026-01-31 20:54:08.608644533 +0100
@@ -6,7 +6,9 @@
 
 #### Tests
 
+if (QT_FEATURE_clang)
 qt_find_package(WrapLibClang 8 PROVIDED_TARGETS WrapLibClang::WrapLibClang)
+endif()
 
 if(TARGET WrapLibClang::WrapLibClang)
     set(TEST_libclang "ON" CACHE BOOL "Required libclang version found." FORCE)
