diff -ur units-2.25.orig/getopt.c units-2.25/getopt.c
--- units-2.25.orig/getopt.c	2026-02-16 13:45:43.399739764 -0500
+++ units-2.25/getopt.c	2026-02-16 13:49:37.180894143 -0500
@@ -175,12 +175,10 @@
 /* Avoid depending on library functions or files
    whose names are inconsistent.  */
 
-char *getenv ();
+char *getenv (const char *name);
 
 static char *
-my_index (str, chr)
-     const char *str;
-     int chr;
+my_index (const char *str, int chr)
 {
   while (*str)
     {
@@ -224,8 +222,7 @@
    the new indices of the non-options in ARGV after they are moved.  */
 
 static void
-exchange (argv)
-     char **argv;
+exchange (char **argv)
 {
   int bottom = first_nonopt;
   int middle = last_nonopt;
@@ -282,8 +279,7 @@
 /* Initialize the internal data when the first call is made.  */
 
 static const char *
-_getopt_initialize (optstring)
-     const char *optstring;
+_getopt_initialize (const char *optstring)
 {
   /* Start processing options with ARGV-element 1 (since ARGV-element 0
      is the program name); the sequence of previously skipped
@@ -372,13 +368,8 @@
    long-named options.  */
 
 int
-_getopt_internal (argc, argv, optstring, longopts, longind, long_only)
-     int argc;
-     char *const *argv;
-     const char *optstring;
-     const struct option *longopts;
-     int *longind;
-     int long_only;
+_getopt_internal (int argc, char *const *argv, const char *optstring,
+		const struct option *longopts, int *longind, int long_only)
 {
   optarg = NULL;
 
@@ -673,10 +664,7 @@
 }
 
 int
-getopt (argc, argv, optstring)
-     int argc;
-     char *const *argv;
-     const char *optstring;
+getopt (int argc, char *const *argv, const char *optstring)
 {
   return _getopt_internal (argc, argv, optstring,
 			   (const struct option *) 0,
@@ -692,9 +680,7 @@
    the above definition of `getopt'.  */
 
 int
-main (argc, argv)
-     int argc;
-     char **argv;
+main (int argc, char **argv)
 {
   int c;
   int digit_optind = 0;
diff -ur units-2.25.orig/getopt.h units-2.25/getopt.h
--- units-2.25.orig/getopt.h	2026-02-16 13:45:43.396847397 -0500
+++ units-2.25/getopt.h	2026-02-16 14:01:06.281087353 -0500
@@ -101,14 +101,7 @@
 #define optional_argument	2
 
 #if defined (__STDC__) && __STDC__
-#ifdef __GNU_LIBRARY__
-/* Many other libraries have conflicting prototypes for getopt, with
-   differences in the consts, in stdlib.h.  To avoid compilation
-   errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int argc, char *const *argv, const char *shortopts);
-#else /* not __GNU_LIBRARY__ */
-extern int getopt ();
-#endif /* __GNU_LIBRARY__ */
 extern int getopt_long (int argc, char *const *argv, const char *shortopts,
 		        const struct option *longopts, int *longind);
 extern int getopt_long_only (int argc, char *const *argv,
diff -ur units-2.25.orig/units.c units-2.25/units.c
--- units-2.25.orig/units.c	2026-02-16 13:45:43.399739764 -0500
+++ units-2.25/units.c	2026-02-16 13:56:13.634635119 -0500
@@ -30,7 +30,7 @@
 
 #define _XOPEN_SOURCE 600
 
-#if defined (_WIN32) && defined (_MSC_VER)
+#if defined (_WIN32)
 # include <windows.h>
 # include <winbase.h>
 # include <wincon.h>
